### DISCUSSION TITLE
How to configure credentials

### DISCUSSION THREAD

**Discussion by LvffY** [renovatebot/renovate#30539] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Azure Devops

## Problem

I'm currently trying to update a repository with some Ansible playbooks.

These playbooks are using some collection that is hosted on a private git repository. 

To be clear, in my Ansible repository, I have a **requirements.yml** file this

```yaml
collections:
  - name: my_git_repo_url
    type: git
    version: 1.0.30
```

When I'm running renovate interactively, renovate asks me for my git password. If I enter my git password, it works.

Now the tricky part is when I want to run the bot non interactively, it stills ask me for the password, but I can't give it.

I tried multiple **hostRules** trying to tell Renovate which token to use but wasn't able to find the right configuration. Am I missing some documentation on how to define this configuration ? 

## Workaround

As of now, I'm using a more low-level git configuration based on [**GIT_ASKPASS** environment variable](https://git-scm.com/docs/gitcredentials/2.29.0).

Which ends to : 

- a **git_askpass.sh** script

```bash
#!/usr/bin/env sh

## This is a script to authenticate through Git for Ansible galaxy process
## We didn't find any other method to authenticate, so this is a little hacky, but this works.

echo $YOUR_ENV_VAR_NAME
```

- Define your git token in **YOUR_ENV_VAR_NAME** environment variable
- Define the **path** to your **git_askpass.sh** script in **GIT_ASKPASS** environment variable

Then run Renovate bot and everything runs fine 👍 

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by mwiede** [comment]
I landed here because I had the same problem, but in the end I missed the correct hostRule. If you activate LOG_DEBUG=debug you should see something like this:
`DEBUG: Found hostRules username and password for url https://foobar.git (repository=xyz/foobar)`
